src/menu/rendersettingsmenustate.cpp

```diff
@@ -369,54 +383,49 @@ std::unique_ptr<MenuState>
       "%1%x Anisotropic Filtering", world.getEngine().getEngineConfig()->renderSettings.anisotropyLevel));
   }
 
-  {
-    int maxW = 0;
-    int totalH = 0;
-    static constexpr int Separation = 10;
-    for(const auto& listBox : m_listBoxes)
-    {
-      listBox->fitToContent();
-      maxW = std::max(maxW, listBox->getSize().x);
-      totalH += listBox->getSize().y + Separation;
-    }
-
-    int y = ui.getSize().y - totalH - 90;
-    for(const auto& listBox : m_listBoxes)
-    {
-      const auto height = listBox->getSize().y;
-      listBox->setSize({maxW, height});
-      listBox->setPosition({(ui.getSize().x - maxW) / 2, y});
-      y += height + Separation;
-    }
-  }
+  m_tabs->fitToContent();
+  m_tabs->setPosition({(ui.getSize().x) / 2 - (m_tabs->getSize().x / 2), (ui.getSize().y / 2) - 90});
 
   for(size_t i = 0; i < m_listBoxes.size(); ++i)
   {
     const auto& listBox = m_listBoxes[i];
     listBox->update(i == m_currentListBox);
-    listBox->draw(ui, world.getPresenter());
   }
+  m_tabs->update(true);
+  m_tabs->draw(ui, world.getPresenter());
 
   const auto& listBox = m_listBoxes[m_currentListBox];
   if(world.getPresenter().getInputHandler().getInputState().zMovement.justChangedTo(hid::AxisMovement::Forward))
   {
     if(!listBox->prevEntry() && m_currentListBox > 0)
+    {
       --m_currentListBox;
+      m_tabs->prevTab();
+    }
   }
   else if(world.getPresenter().getInputHandler().getInputState().zMovement.justChangedTo(hid::AxisMovement::Backward))
   {
     if(!listBox->nextEntry() && m_currentListBox < m_listBoxes.size() - 1)
+    {
       ++m_currentListBox;
+      m_tabs->nextTab();
+    }
   }
   if(world.getPresenter().getInputHandler().getInputState().xMovement.justChangedTo(hid::AxisMovement::Left))
   {
     if(m_currentListBox > 0)
+    {
       --m_currentListBox;
+      m_tabs->prevTab();
+    }
   }
   else if(world.getPresenter().getInputHandler().getInputState().xMovement.justChangedTo(hid::AxisMovement::Right))
```

Comment:
please use step left/right for moving between the tabs. once the slider widget will be added, plain left/right will be moved to change the slider value. also, you may want to change the input mapping widget to use the tab widget instead, which already uses step left/right to switch between input mappings.